QEfficient/transformers/quantizers/quant_transforms.py

```diff
@@ -0,0 +1,51 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import torch
+from torch import nn
+
+from QEfficient.base.pytorch_transforms import ModuleMutatorTransform
+from QEfficient.customop.matmulnbits import QuantLinearORT
+from QEfficient.transformers.quantizers.awq import WQLinear_GEMM, unpack_awq_weights
+
+
+class AwqToMatmulNbitsTransform(ModuleMutatorTransform):
+    _match_class = WQLinear_GEMM
+
+    @staticmethod
+    def unpack_and_dequantize_awq(qweight, qzeros, scales, bits, group_size):
+        # Unpack the qweight and qzeros tensors
+        scales, int_weight, int_zeros = unpack_awq_weights(qweight, qzeros, scales, bits)
+
+        # fp16 weights
+        scales_expand = scales.repeat_interleave(group_size, dim=0)
+        int_zeros_expand = int_zeros.repeat_interleave(group_size, dim=0)
+        int_weight = (int_weight - int_zeros_expand) * scales_expand
+
+        return int_weight.T, scales, int_zeros.to(torch.int32)
+
+    @classmethod
+    def mutate(cls, original_module: nn.Module, parent_module: nn.Module):
+        fp16_weight, scales, zeros = cls.unpack_and_dequantize_awq(
+            original_module.qweight,
+            original_module.qzeros,
+            original_module.scales,
+            original_module.w_bit,
+            original_module.group_size,
+        )
+
+        original_module.weight = fp16_weight
+        new_module = QuantLinearORT(
+            original_module.w_bit,
+            original_module.group_size,
+            original_module.in_features,
+            original_module.out_features,
+            original_module.bias is not None,
+        )
+        new_module.bias = original_module.bias if original_module.bias is not None else None
+        new_module.pack(original_module, scales.T, zeros.T, original_module.g_idx)
```

Comment:
Here we have to call a method of new_module `new_module.pack`, we could simply move the `unpack_and_dequantize` also into that class and call these 2 from `__qeff_init__`.

This will make the code look simpler and not scattered across multiple files for 1 feature.

Reply:
The `upack_and_dequantize` code will be moved to `quantized_utils.py` as They `awq` and `gptq` share same code along with their own differences.
It is possible to move this code to a base class that both GPTQ and AWQ linear classes inherit.
I will check the best possible way to do this.